meerk40t/gui/panes/navigationpanels.py

```diff
@@ -872,24 +944,43 @@ def __do_layout(self):
         grid_sizer_2.Add(self.button_rotate_ccw, 0, 0, 0)
         grid_sizer_2.Add(self.button_translate_down, 0, 0, 0)
         grid_sizer_2.Add(self.button_rotate_cw, 0, 0, 0)
-        grid_sizer_2.Add(self.text_a, 0, 0, 0)
-        grid_sizer_2.Add(self.text_c, 0, 0, 0)
-        grid_sizer_2.Add(self.text_e, 0, 0, 0)
-        grid_sizer_2.Add(self.text_b, 0, 0, 0)
-        grid_sizer_2.Add(self.text_d, 0, 0, 0)
-        grid_sizer_2.Add(self.text_f, 0, 0, 0)
+
+        grid_sizer_3 = wx.FlexGridSizer(3, 4, 0, 0)
+        # Add some labels to make textboxes clearer to understand
+        grid_sizer_3.Add(wx.StaticText(self, wx.ID_ANY, ""), wx.HORIZONTAL)
+        grid_sizer_3.Add(wx.StaticText(self, wx.ID_ANY, _("Scale")), wx.HORIZONTAL)
+        grid_sizer_3.Add(wx.StaticText(self, wx.ID_ANY, _("Skew")), wx.HORIZONTAL)
+        grid_sizer_3.Add(wx.StaticText(self, wx.ID_ANY, _("Translate")), wx.HORIZONTAL)
+        # First X
+        grid_sizer_3.Add(wx.StaticText(self, wx.ID_ANY, _("X:")), wx.HORIZONTAL)
+        grid_sizer_3.Add(self.text_a, 0, 0, 0)  # Scale X
+        grid_sizer_3.Add(self.text_c, 0, 0, 0)  # Skew X
+        grid_sizer_3.Add(self.text_e, 0, 0, 0)  # Translate X
+        # Then Y
+        grid_sizer_3.Add(wx.StaticText(self, wx.ID_ANY, _("Y:")), wx.HORIZONTAL)
+        grid_sizer_3.Add(self.text_d, 0, 0, 0)  # Scale Y
+        grid_sizer_3.Add(self.text_b, 0, 0, 0)  # Skew Y
```

Comment:
I think this needs to come before text_d.

Reply:
We need to take a decision here:
A) If we want to mirror the SVG matrix definition then the fields need to be:
SC_X SK_X TL_X
SK_Y SC_Y TL_Y
(as the wey were previously) - then we should add another row below Y to show "Skew Scale Translate" or omit the headers row to avoid confusion
B) If we want to hide the internal matrix representation and make the fields loook more 'logically' to your average Joe/Jane (and where the majority mous probably has no clue about SVG matrices in the first place, and I count myself to that group as of a couple of days ago...) then we should keep the representation as in this PR

Well, my preference is B)   ;-)

Reply:
My vote is 👍 for option B).

Reply:
Well, the fields match the SVG provision. We don't *really* expose this fact, the `matrix` command is, after all a console command. It's fairly well hidden. The text boxes in navigation give that information but it doesn't much matter a it usually doesn't come up. The SVG style for the matrixes matches the wxPython style for the matrixes. For example when you pinch and zoom in the scene, that's using the SVG style matrix to do that and transfers that over to wxPython matrix things in the zmatrix or whatever that is, and it's just the same values in the same order.

Reply:
There's not really that much of a *logical* order to those values if they aren't a direct rotation or have a skew or something or sometimes are scaled and rotated. The values of the matrix are sometimes helpful to know but mostly you'd need to make some guesses when decomposing the matrix into a series of scale/rotate/translate operations. Your skew values for angles get a little weird when you're moving the points around, and you just can't quite read it as independent operations in a way that is helpful even when you know the math behind it.

Reply:
I agree the matrix gets pretty weird soon, still it has become my favourite tool to exactly place / scale something (I do a lot of on-the-fly rects...) and there it helps :-)